### ISSUE TITLE
Step: compare YAML files

### ISSUE THREAD

**Issue by plocket** [SuffolkLITLab/ALKiln#470]
### Feedback needed for behavior

1. Where should the comparison YAML files be stored? React with ❤️ for the Sources folder. React with 👍 for a separate folder on the repo itself. For other ideas, add a comment. The developer will use the name of the comparison file in the Step.
2. Should the downloaded YAML files be made into artifacts only if they error or always? React with 😄 for only if they error, react with 🎉 for always.

Requested for ALWeaver refactoring regression tests.

**Step text:**

```
And "file_url_name.yml" matches "original_filename.yml"
When I download "file_url_name.yml" it matches "original_filename.yml"
```
~[I'm planning on option 2 at the moment]~

~Should it be `matches` or `is the same as`? Something else?~

Wanted to highlight that it needs to be a text value, so went with
```
Then the text in the JSON variable "variable_with_text_value" should be
```

### Other logistics

**YAML comparison:**

I've looked around and https://www.npmjs.com/package/js-yaml seems like the hottest kid on the block.

Theoretically, we might be able to just compare the YAML files as text, but then superficial changes like whitespace fixes would require a new file uploaded. Comparing JSON will be more flexible. We can use [chai's `.deep.equal()`](https://www.chaijs.com/api/bdd/#method_deep) to check for equality.

Someday we can think of adding a pretty diff, but I had trouble finding a good library. For now we might leave instructions to use a diff checker.

**Renaming files:**

We can only do it in a kind of annoying and gnarly way - download the file, detect the completion, rename that specific file after it's downloaded
- Get current filename so we can change it after: https://stackoverflow.com/a/57419156/14144258. Not sure what we'll do if the filename the user gives to rename to is the same as another file that's currently in the folder.
- Rename the file: https://stackoverflow.com/a/22504722/14144258

**Comment by plocket** [comment]
React to this with 👍 if a filename should automatically append its page id regardless of the name the user gives to it.

**Comment by plocket** [comment]
For YAML and this situation specifically, we might make a totally different step. Maybe:

```
Then the value of the JSON variable "var_name" should match "file_name.txt/json/something requirable"
```

> It looks like the interview source is in a template, not directly a variable you could grab. https://github.com/SuffolkLITLab/docassemble-ALWeaver/blob/e7a83717e0079f16a2fae5b18a54b079b2f255f1/docassemble/ALWeaver/data/questions/assembly_line.yml#L1420-L1426. It would be a 1 line fix to save the string as a variable though:
> ```py
> interview_source = interview.source()
> ```

This way we don't have to deal with downloading and then opening files yet, which is a whole hassle.

**Comment by plocket** [comment]
Since we're just comparing strings, maybe devs can just write a cucumber string.

```
    Then the value of the JSON variable "var_name" should be
    """
    metadata:
      key1: "value"
    """
```

We might end up making multiple test files for that because it would be a bit messy, but we'd otherwise create multiple YAML files. The YAML files could be in a separate non-da folder, though. Not sure which would be better.

**Comment by plocket** [comment]
React with 👍 if the downloaded JSON variables data should be stored as a `.json`. Reach with 😄 for a `.txt`. The latter would be more accessible to more people, I think. Regular text editors don't open `.json` files, right?

**Comment by plocket** [comment]
Closed by #479 and #481
